addons/l10n_eg_edi_eta/models/account_move.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+import base64
+import hashlib
+import hmac
+import time
+from collections import defaultdict
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+import logging
+
+from odoo.tools import formatLang
+from odoo.tools.float_utils import float_compare, float_is_zero, float_round
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_eg_long_id = fields.Char('ETA Long ID', copy=False)
+    l10n_eg_internal_id = fields.Char('ETA Internal ID', copy=False)
+    l10n_eg_hash_key = fields.Char('ETA Hash Key', copy=False)
+    l10n_eg_uuid = fields.Char(string='Document UUID', copy=False)
+    l10n_eg_submission_id = fields.Char(string='Submission ID', copy=False)
+    l10n_eg_signature_type = fields.Char(string='Signature Type', copy=False)
+    l10n_eg_signature_data = fields.Text('Signature Data', copy=False)
+    l10n_eg_posted_datetime = fields.Datetime('Posted Date', copy=False)
```

Comment:
There seems to be no need for this field.  It might be better to use the datetime.now() when signing.  The only problem is that we would need to store the json we send and then just add the signature. (instead of regenerating which is always tricky)  All of these fields above can be computed if we put the response in the json file  (maybe we can do the json as {'request': {}, 'response': {}}, but only when successful and then return it as attachment.